apps/native/app/src/screens/document-detail/document-reply.tsx

```diff
@@ -0,0 +1,241 @@
+
+import styled from 'styled-components/native'
+import { LoadingIcon } from '../../components/nav-loading-spinner/loading-icon'
+import { Pressable } from '../../components/pressable/pressable'
+import { useUser } from '../../contexts/user-provider'
+import { useDocumentReplyMutation } from '../../graphql/types/schema'
+import { createNavigationOptionHooks } from '../../hooks/create-navigation-option-hooks'
+import { useNavigationModal } from '../../hooks/use-navigation-modal'
+import { useAuthStore } from '../../stores/auth-store'
+import {
+  Button,
+  Container,
+  Icon,
+  NavigationBarSheet,
+  Problem,
+  Spacing,
+  TextField,
+  Typography,
+} from '../../ui'
+import { ComponentRegistry } from '../../utils/component-registry'
+import { isAndroid } from '../../utils/devices'
+
+const Host = styled.SafeAreaView`
+  flex: 1;
+`
+
+const Row = styled(Container)<{
+  hasBottomBorder?: boolean
+  paddingVertical?: Spacing
+}>(({ hasBottomBorder = true, paddingVertical = 1, theme }) => ({
+  flexDirection: 'row',
+  paddingTop: theme.spacing[paddingVertical],
+  paddingBottom: theme.spacing[paddingVertical],
+  ...(hasBottomBorder && {
+    borderBottomWidth: theme.border.width.standard,
+    borderBottomColor: theme.color.blue200,
+  }),
+}))
+
+const HeaderTitle = styled(Row)`
+  padding-bottom: ${({ theme }) => theme.spacing.p2}px;
+`
+
+const TextAreaWrapper = styled.View`
+  flex: 1;
+`
+
+const TextArea = styled(TextField)`
+  height: 170px;
+`
+
+const Footer = styled(Container)(({ theme }) => ({
+  flex: 1,
+  justifyContent: 'flex-end',
+  ...(isAndroid && {
+    paddingBottom: theme.spacing.p2,
+  }),
+}))
+
+const EmailIconWrapper = styled(Pressable)`
+  margin-left: auto;
+`
+
+const { getNavigationOptions, useNavigationOptions } =
+  createNavigationOptionHooks(() => ({
+    topBar: {
+      visible: false,
+    },
+  }))
+
+export type DocumentReplyScreenProps = {
+  senderName: string
+  documentId: string
+  subject: string
+  isFirstReply?: boolean
+  onReplySuccess?(showFirstReplyInfo: boolean): void
+}
+
+export const DocumentReplyScreen: NavigationFunctionComponent<
+  DocumentReplyScreenProps
+> = ({
+  componentId,
+  senderName,
+  documentId,
+  subject,
+  isFirstReply = false,
+  onReplySuccess,
+}) => {
+  useNavigationOptions(componentId)
+  const intl = useIntl()
+  const { showModal, dismissModal } = useNavigationModal()
+  const { userInfo } = useAuthStore()
+  const { user: userProfile, loading, error } = useUser()
+
+  const [message, setMessage] = useState('')
+  const [hasShownModal, setHasShownModal] = useState(false)
+
+  const [sendMessage, { loading: sendMessageLoading }] =
+    useDocumentReplyMutation({
+      onCompleted: (data) => {
+        const id = data.documentsV2Reply?.id
+
+        if (id) {
+          // Successful reply, clear message
+          setMessage('')
+          onReplySuccess?.(isFirstReply)
+          dismissModal(componentId)
```

Comment:
For the first reply, the modal is on top of document-detail and needs to  somehow navigate to the communication screen with proper first reply props.